diesel/src/pg/types/date_and_time/mod.rs

```diff
@@ -44,12 +44,21 @@ pub struct PgTime(pub i64);
 /// meaning of these parts.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub struct PgInterval {
+    /// The number of whole microseconds
     pub microseconds: i64,
+    /// The number of whole days
     pub days: i32,
+    /// The number of whole months
     pub months: i32,
 }
 
 impl PgInterval {
+    /// Constructs a new `PgInterval`
+    ///
+    /// No conversion occurs on the arguments. It is valid to provide a number
+    /// of microseconds greater than the longest possible day, or a number of
+    /// days greater than the longest possible month, as it is impossible to say
+    /// how many months are in "40 days" without knowing a precise date.
```

Comment:
Maybe also add an answer to: What happens when this kind of invalid data is used in a query?